apis/network/v1alpha3/register.go

```diff
@@ -53,7 +53,17 @@ var (
 	SubnetGroupVersionKind = SchemeGroupVersion.WithKind(SubnetKind)
 )
 
+// SecurityGroup type Metadata.
+var (
+	SecurityGroupKind             = reflect.TypeOf(SecurityGroup{}).Name()
+	SecurityGroupGroupKind        = schema.GroupKind{Group: Group, Kind: SecurityGroupKind}.String()
+	SecurityGroupKindAPIVersion   = SecurityGroupKind + "." + SchemeGroupVersion.String()
+	SecurityGroupGroupVersionKind = SchemeGroupVersion.WithKind(SecurityGroupKind)
+)
+
 func init() {
 	SchemeBuilder.Register(&VirtualNetwork{}, &VirtualNetworkList{})
 	SchemeBuilder.Register(&Subnet{}, &SubnetList{})
+	SchemeBuilder.Register(&SecurityGroup{}, &SecurityGroupList{})
+	SchemeBuilder.Register(&SecurityRule{})
```

Comment:
Guessing there should also be a `SecurityRuleList`?

Reply:
Or is it intended that a `SecurityRule` just be in the spec of a `SecurityGroup`? If so, it should not be its own object, but it they are separate APIs on Azure then they should be separate Crossplane objects and the `SecurityGroup` should reference `SecurityRule`.

Reply:
Refactored SecurityRule as it is just a part of Security Group not a independent entity.